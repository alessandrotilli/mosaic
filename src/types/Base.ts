export interface IBase {
  dataCy?: string;
}

export interface ILoadable extends IBase {
  loading?: boolean;
}

// TODO: door open to allow explicit property localization
export interface ILocalizable extends IBase {
  localized?: boolean;
}
