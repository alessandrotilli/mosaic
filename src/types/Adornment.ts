import { IBase } from "./Base";

type IBadgeColor = "default" | "error" | "primary" | "secondary";

type IBadgeVariant = "dot" | "standard";

export interface IBadgeAdornment {
  color?: IBadgeColor;
  value: string;
  variant?: IBadgeVariant;
}

export interface IAdornment extends IBase {
  /**
   * Badge properties
   */
  badge?: IBadgeAdornment;
  /**
   * Text of tooltip
   */
  tooltip?: string;
}
