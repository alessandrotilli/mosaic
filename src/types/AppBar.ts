import { ILocalizable } from "./Base";
import { IBaseButton } from "./Button";
import { IBaseIconButton } from "./IconButton";

export interface IAppBar extends ILocalizable {
  actions?: IBaseIconButton[];
  menu?: IBaseIconButton;
  onTitleClick?: () => void;
  title?: string;
  userMenu?: IBaseButton[];
  username?: string;
}
