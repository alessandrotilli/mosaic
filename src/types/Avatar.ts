import { ILoadable } from "./Base";
import { Icons } from "./Icon";

export enum AvatarVariant {
  default = "circular",
  rounded = "rounded",
  squared = "square",
}

export interface IAvatar extends ILoadable {
  alt?: string;
  icon?: Icons;
  src?: string;
  text?: string;
  variant?: AvatarVariant;
}
