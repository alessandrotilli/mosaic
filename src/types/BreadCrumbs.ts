import { IBase } from "./Base";

export type BreadCrumbsType = IBase & {
  link: { label: string; href: string }[];
  onClick: (href: string) => void;
};
