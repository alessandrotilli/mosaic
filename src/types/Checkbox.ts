import { ILocalizable } from "./Base";

export enum CheckboxSize {
  small = "small",
  default = "medium",
}

export interface ICheckbox extends ILocalizable {
  disabled?: boolean;
  intermediate?: boolean;
  label?: string;
  labelPlacement?: "end" | "start";
  onChange?: (checked: boolean) => void;
  required?: boolean;
  size?: CheckboxSize;
  value?: boolean;
}
