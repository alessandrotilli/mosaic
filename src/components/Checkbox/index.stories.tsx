import React from "react";
import { action } from "@storybook/addon-actions";
import { boolean, select, text } from "@storybook/addon-knobs";

import { CheckboxSize } from "../../types/Checkbox";
import FormMock from "../../utils/mocks/FormMock";
import { getDocumentationPage, StoriesWrapper } from "../../utils/stories";

import Checkbox, { DATA_CY_DEFAULT } from ".";

export default {
  title: "Checkbox",
  component: Checkbox,
  parameters: {
    ...getDocumentationPage({
      basedOn: "@material-ui/core/Checkbox",
      component: "Checkbox",
      e2eTestInfo: {
        dataCyDefault: DATA_CY_DEFAULT,
      },
    }),
  },
};

export const Canvas = () => (
  // FormMock simulates external form component handling state
  // In a real case scenario "onChange" and "value" props must be passed to Checkbox
  <FormMock inputValue={boolean("value", true)} onInputChange={action("Change checkbox")}>
    <Checkbox
      dataCy={text("data-cy", "checkbox")}
      disabled={boolean("disabled", false)}
      intermediate={boolean("intermediate", false)}
      onChange={action("Change checkbox")}
      required={boolean("value", false)}
      size={select("size", CheckboxSize, CheckboxSize.small, CheckboxSize.default)}
      value={boolean("value", true)}
    />
  </FormMock>
);

export const Disabled = () => (
  <StoriesWrapper>
    <Checkbox dataCy={"checkbox"} value={false} disabled />
    <Checkbox dataCy={"checkbox"} value={true} disabled />
    <Checkbox dataCy={"checkbox"} intermediate disabled />
  </StoriesWrapper>
);

export const Size = () => (
  <StoriesWrapper>
    <Checkbox dataCy={"checkbox-identifier"} value={true} />
    <Checkbox dataCy={"checkbox-identifier"} value={true} size={CheckboxSize.small} />
    <Checkbox dataCy={"checkbox-identifier"} value={false} />
    <Checkbox dataCy={"checkbox-identifier"} value={false} size={CheckboxSize.small} />
  </StoriesWrapper>
);

export const Values = () => (
  <StoriesWrapper>
    <Checkbox dataCy={"checkbox-identifier"} value={false} onChange={() => {}} />
    <Checkbox dataCy={"checkbox-identifier"} value={true} onChange={() => {}} />
    <Checkbox dataCy={"checkbox-identifier"} value={false} intermediate onChange={() => {}} />
    <Checkbox dataCy={"checkbox-identifier"} value={true} intermediate onChange={() => {}} />
  </StoriesWrapper>
);
