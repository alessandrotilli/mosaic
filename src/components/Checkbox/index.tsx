import React, { ChangeEvent, FC, useCallback } from "react";
import { Checkbox as MUICheckbox, FormControlLabel as MUIFormControlLabel } from "@mui/material";

import { ICheckbox } from "../../types/Checkbox";
import { getComposedDataCy } from "../../utils";
import localized, { ILocalizableProperty } from "../../utils/hocs/localized";

export const DATA_CY_DEFAULT = "checkbox";
export const DATA_CY_SHORTCUT = "label";
export const LOCALIZABLE_PROPS: ILocalizableProperty[] = [{ name: "label", type: "string" }];

export const SUBPARTS_MAP = {
  check: {
    label: "Check",
  },
  input: {
    label: "Input",
  },
};

const Checkbox: FC<ICheckbox> = ({
  dataCy = DATA_CY_DEFAULT,
  disabled = false,
  intermediate = false,
  label,
  labelPlacement = "start",
  onChange,
  required = false,
  size = "medium",
  style,
  value = false,
}) => {
  const onChangeHandler = useCallback(
    ({ target: { checked } }: ChangeEvent<HTMLInputElement>) => onChange && onChange(checked),
    [onChange]
  );

  return (
    <MUIFormControlLabel
      control={
        <MUICheckbox
          checked={value}
          data-cy={getComposedDataCy(dataCy, SUBPARTS_MAP.check)}
          disabled={disabled}
          indeterminate={intermediate}
          inputProps={
            {
              "data-cy": getComposedDataCy(dataCy, SUBPARTS_MAP.input),
            } as any
          }
          onChange={onChangeHandler}
          required={required}
          size={size}
          style={style}
        />
      }
      data-cy={dataCy}
      label={label}
      labelPlacement={labelPlacement}
      style={{ ...(!label ? { margin: 0 } : {}) }}
    />
  );
};

export const CheckboxWithProps = Checkbox;

export default localized(Checkbox, {
  dataCyShortcut: DATA_CY_SHORTCUT,
  localizableProps: LOCALIZABLE_PROPS,
});
