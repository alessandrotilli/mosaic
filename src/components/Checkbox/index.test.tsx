import React from "react";
import { mount, ReactWrapper } from "enzyme";
import { ICheckbox, CheckboxSize } from "../../types/Checkbox";
import Checkbox, { DATA_CY_DEFAULT } from ".";

const defaultProps: ICheckbox = {};

const getElement = (props?: ICheckbox, dataCy = DATA_CY_DEFAULT): ReactWrapper => {
  const wrapper = mount(<Checkbox {...defaultProps} {...props} />);
  return wrapper.find("span[data-cy='" + dataCy + "']");
};

describe("Checkbox test suite:", () => {
  it("default", () => {
    const element = getElement();
    expect(element).toHaveLength(1);
    const input = element.find("input");
    expect(input.prop("checked")).toBeFalsy();
    expect(input.prop("data-indeterminate")).toBeFalsy();
    expect(input.prop("disabled")).toBeFalsy();
    expect(input.prop("required")).toBeFalsy();
  });

  it("dataCy", () => {
    const element = getElement({ dataCy: "custom" }, "custom");
    expect(element).toHaveLength(1);
  });

  it("checked", () => {
    const element = getElement({ value: true });
    const input = element.find("input");
    expect(input.prop("checked")).toBeTruthy();
  });

  it("disabled", () => {
    const element = getElement({ disabled: true });
    const input = element.find("input");
    expect(input.prop("disabled")).toBeTruthy();
  });

  it("intermediate", () => {
    const element = getElement({ intermediate: true });
    const input = element.find("input");
    expect(input.prop("data-indeterminate")).toBeTruthy();
  });

  it("onChange", () => {
    const onChangeHandler = jest.fn();
    const element = getElement({ onChange: onChangeHandler });
    const inputBefore = element.find("input");
    inputBefore.simulate("change", { target: { checked: true } });
    expect(onChangeHandler).toHaveBeenCalledTimes(1);
    expect(onChangeHandler).toHaveBeenCalledWith(true);
  });

  it("onChange - not handled", () => {
    const element = getElement();
    const inputBefore = element.find("input");
    inputBefore.simulate("change", { target: { checked: true } });
  });

  it("required", () => {
    const element = getElement({ required: true });
    const input = element.find("input");
    expect(input.prop("required")).toBeTruthy();
  });

  it("size", () => {
    const element = getElement({ size: CheckboxSize.small });
  });
});
