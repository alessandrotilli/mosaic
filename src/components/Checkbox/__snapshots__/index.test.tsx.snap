// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox test suite: checked 1`] = `
<label
  className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart"
  data-cy="checkbox"
  style={
    {
      "margin": 0,
    }
  }
>
  <span
    aria-disabled={false}
    className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-checked-2 Mui-checked MuiIconButton-colorPrimary"
    data-cy="checkbox-check"
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={null}
  >
    <span
      className="MuiIconButton-label"
    >
      <input
        checked={true}
        className="PrivateSwitchBase-input-4"
        data-cy="checkbox-input"
        data-indeterminate={false}
        disabled={false}
        onChange={[Function]}
        required={false}
        type="checkbox"
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </span>
  <span
    className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
  />
</label>
`;

exports[`Checkbox test suite: dataCy 1`] = `
<label
  className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart"
  data-cy="custom"
  style={
    {
      "margin": 0,
    }
  }
>
  <span
    aria-disabled={false}
    className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
    data-cy="custom-check"
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={null}
  >
    <span
      className="MuiIconButton-label"
    >
      <input
        checked={false}
        className="PrivateSwitchBase-input-4"
        data-cy="custom-input"
        data-indeterminate={false}
        disabled={false}
        onChange={[Function]}
        required={false}
        type="checkbox"
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </span>
  <span
    className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
  />
</label>
`;

exports[`Checkbox test suite: default 1`] = `
<label
  className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart"
  data-cy="checkbox"
  style={
    {
      "margin": 0,
    }
  }
>
  <span
    aria-disabled={false}
    className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
    data-cy="checkbox-check"
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={null}
  >
    <span
      className="MuiIconButton-label"
    >
      <input
        checked={false}
        className="PrivateSwitchBase-input-4"
        data-cy="checkbox-input"
        data-indeterminate={false}
        disabled={false}
        onChange={[Function]}
        required={false}
        type="checkbox"
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
    </span>
  </span>
  <span
    className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
  />
</label>
`;

exports[`Checkbox test suite: disabled 1`] = `
<label
  className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart Mui-disabled"
  data-cy="checkbox"
  style={
    {
      "margin": 0,
    }
  }
>
  <span
    aria-disabled={true}
    className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-disabled-3 Mui-disabled MuiIconButton-colorPrimary Mui-disabled Mui-disabled"
    data-cy="checkbox-check"
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={-1}
  >
    <span
      className="MuiIconButton-label"
    >
      <input
        checked={false}
        className="PrivateSwitchBase-input-4"
        data-cy="checkbox-input"
        data-indeterminate={false}
        disabled={true}
        onChange={[Function]}
        required={false}
        type="checkbox"
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
    </span>
  </span>
  <span
    className="MuiTypography-root MuiFormControlLabel-label Mui-disabled MuiTypography-body1"
  />
</label>
`;

exports[`Checkbox test suite: intermediate 1`] = `
<label
  className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart"
  data-cy="checkbox"
  style={
    {
      "margin": 0,
    }
  }
>
  <span
    aria-disabled={false}
    className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-indeterminate MuiIconButton-colorPrimary"
    data-cy="checkbox-check"
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={null}
  >
    <span
      className="MuiIconButton-label"
    >
      <input
        checked={false}
        className="PrivateSwitchBase-input-4"
        data-cy="checkbox-input"
        data-indeterminate={true}
        disabled={false}
        onChange={[Function]}
        required={false}
        type="checkbox"
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-2 10H7v-2h10v2z"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </span>
  <span
    className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
  />
</label>
`;

exports[`Checkbox test suite: label 1`] = `
<label
  className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart"
  data-cy="checkbox"
  style={{}}
>
  <span
    aria-disabled={false}
    className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
    data-cy="checkbox-check"
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={null}
  >
    <span
      className="MuiIconButton-label"
    >
      <input
        checked={false}
        className="PrivateSwitchBase-input-4"
        data-cy="checkbox-input"
        data-indeterminate={false}
        disabled={false}
        onChange={[Function]}
        required={false}
        type="checkbox"
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
    </span>
  </span>
  <span
    className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
  >
    Checkbox
  </span>
</label>
`;

exports[`Checkbox test suite: label placement 1`] = `
<label
  className="MuiFormControlLabel-root"
  data-cy="checkbox"
  style={{}}
>
  <span
    aria-disabled={false}
    className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
    data-cy="checkbox-check"
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={null}
  >
    <span
      className="MuiIconButton-label"
    >
      <input
        checked={false}
        className="PrivateSwitchBase-input-4"
        data-cy="checkbox-input"
        data-indeterminate={false}
        disabled={false}
        onChange={[Function]}
        required={false}
        type="checkbox"
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </span>
  <span
    className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
  >
    Checkbox
  </span>
</label>
`;

exports[`Checkbox test suite: localized 1`] = `
<label
  className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart"
  data-cy="locale.checkbox"
  style={{}}
>
  <span
    aria-disabled={false}
    className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
    data-cy="locale.checkbox-check"
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={null}
  >
    <span
      className="MuiIconButton-label"
    >
      <input
        checked={false}
        className="PrivateSwitchBase-input-4"
        data-cy="locale.checkbox-input"
        data-indeterminate={false}
        disabled={false}
        onChange={[Function]}
        required={false}
        type="checkbox"
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
    </span>
  </span>
  <span
    className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
  >
    Checkbox
  </span>
</label>
`;

exports[`Checkbox test suite: onChange - not handled 1`] = `
<label
  className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart"
  data-cy="checkbox"
  style={
    {
      "margin": 0,
    }
  }
>
  <span
    aria-disabled={false}
    className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
    data-cy="checkbox-check"
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={null}
  >
    <span
      className="MuiIconButton-label"
    >
      <input
        checked={false}
        className="PrivateSwitchBase-input-4"
        data-cy="checkbox-input"
        data-indeterminate={false}
        disabled={false}
        onChange={[Function]}
        required={false}
        type="checkbox"
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </span>
  <span
    className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
  />
</label>
`;

exports[`Checkbox test suite: onChange 1`] = `
<label
  className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart"
  data-cy="checkbox"
  style={
    {
      "margin": 0,
    }
  }
>
  <span
    aria-disabled={false}
    className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
    data-cy="checkbox-check"
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={null}
  >
    <span
      className="MuiIconButton-label"
    >
      <input
        checked={false}
        className="PrivateSwitchBase-input-4"
        data-cy="checkbox-input"
        data-indeterminate={false}
        disabled={false}
        onChange={[Function]}
        required={false}
        type="checkbox"
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
    </span>
  </span>
  <span
    className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
  />
</label>
`;

exports[`Checkbox test suite: required 1`] = `
<label
  className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart"
  data-cy="checkbox"
  style={
    {
      "margin": 0,
    }
  }
>
  <span
    aria-disabled={false}
    className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
    data-cy="checkbox-check"
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={null}
  >
    <span
      className="MuiIconButton-label"
    >
      <input
        checked={false}
        className="PrivateSwitchBase-input-4"
        data-cy="checkbox-input"
        data-indeterminate={false}
        disabled={false}
        onChange={[Function]}
        required={true}
        type="checkbox"
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
    </span>
  </span>
  <span
    className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
  />
</label>
`;

exports[`Checkbox test suite: size 1`] = `
<label
  className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart"
  data-cy="checkbox"
  style={
    {
      "margin": 0,
    }
  }
>
  <span
    aria-disabled={false}
    className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
    data-cy="checkbox-check"
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={null}
  >
    <span
      className="MuiIconButton-label"
    >
      <input
        checked={false}
        className="PrivateSwitchBase-input-4"
        data-cy="checkbox-input"
        data-indeterminate={false}
        disabled={false}
        onChange={[Function]}
        required={false}
        type="checkbox"
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </span>
  <span
    className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
  />
</label>
`;
